plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -62,6 +62,57 @@ public function register_routes() {
 				'schema' => array( $this, 'get_public_item_schema' ),
 			)
 		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)/duplicate',
+			array(
+				'args'   => array(
+					'id' => array(
+						'description' => __( 'Unique identifier for the resource.', 'woocommerce' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'duplicate_product' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Duplicate a product and returns the duplicated product.
+	 * The product status is set to "draft" and the name includes a "(copy)" at the end by default.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function duplicate_product( $request ) {
+		$product_id = $request->get_param( 'id' );
+		$product    = wc_get_product( $product_id );
+
+		if ( ! $product ) {
+			return new WP_Error( 'woocommerce_rest_product_invalid_id', __( 'Invalid product ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		if ( 'simple' !== $product->get_type() ) {
+			$request['type'] = $product->get_type();
+		}
+
+		$updated_product    = $this->prepare_object_for_database( $request );
+		$duplicated_product = ( new WC_Admin_Duplicate_Product() )->product_duplicate( $updated_product );
```

Comment:
The use of `prepare_object_for_database( $request )` to get a product object that we than copy is clever, but also feels a bit weird to me.
Is this something we have done in the past? 

I am ok with leaving this, but maybe just adding an inline comment would be helpful, something like:
`// Creating product object from request data in preparation for copying.`

Reply:
Yeah, it seems like a naming problem.
I'm good with adding the comment, but if you prefer, I can add a parameter to `prepare_object_for_database` and use the `( new WC_Admin_Duplicate_Product() )->product_duplicate( $product )` inside.

I'm leaning towards the first solution (adding a comment) because the duplication at the end is a way to save the product (slightly modified), so it would keep the pattern of `preparing then saving` the product.

Reply:
@louwie17 I just added the comment in the commit [290345e](https://github.com/woocommerce/woocommerce/pull/46141/commits/290345e41d13baacde614ff4164aa512c5954619)